File: template/app/src/main/java/com/xmartlabs/template/ui/common/TemplateView.java
package com.xmartlabs.template.ui.common;

import android.content.Context;
import android.support.annotation.StringRes;

import com.xmartlabs.bigbang.ui.mvp.MvpView;

import io.reactivex.annotations.NonNull;

public interface TemplateView extends MvpView {
  void showError(@StringRes int message);
  void showError(@StringRes int message, @StringRes int title);
  void showError(@StringRes int message, @StringRes int title, @StringRes int buttonTitle);
  void showError(@NonNull Throwable error, @StringRes Integer message);
  String getString(@StringRes int stringRes);
  boolean isViewAlive();
  Context getContext();
}

File: template/app/src/main/java/com/xmartlabs/template/ui/common/TemplateView.kt
package com.xmartlabs.template.ui.common

import android.support.annotation.StringRes

import com.xmartlabs.bigbang.ui.mvp.MvpView
import com.xmartlabs.template.R

interface TemplateView : MvpView {
  fun setup()
  fun showError(@StringRes message: Int, @StringRes title: Int = R.string.error,
                @StringRes buttonTitle: Int = android.R.string.ok)
  fun showError(error: Throwable, @StringRes message: Int?)
  val isViewAlive: Boolean
}
